client/boot/index.js

```diff
@@ -228,6 +231,26 @@ function reduxStoreReady( reduxStore ) {
 	// Layout themselves when logged in.
 	if ( ! isIsomorphic || user.get() ) {
 		renderLayout( reduxStore );
+
+		if ( config.isEnabled( 'catch-js-errors' ) ) {
+			const Logger = require( 'lib/catch-js-errors' );
+			const errorLogger = new Logger();
+			//Save data to JS error logger
+			errorLogger.saveDiagnosticData( {
+				user_id: user.get().ID,
+				calypso_env: config( 'env_id' )
+			} );
+			errorLogger.saveDiagnosticReducer( function() {
+				const state = reduxStore.getState();
+				return {
+					blog_id: getSelectedSiteId( state ),
+					calypso_section: getSectionName( state )
+				};
+			} );
+			errorLogger.saveDiagnosticReducer( () => ( { tests: getSavedVariations() } ) );
+			analytics.on( 'record-event', ( eventName, eventProperties ) => errorLogger.saveExtraData( { lastTracksEvent: eventProperties } ) );
+			analytics.on( 'page-view', ( urlPath ) => errorLogger.saveDiagnosticData( { calypso_path: urlPath } ) );
```

Comment:
I did it in the same way, but that would require implementing my own logic separating siteSlug from path.
Analytics already has this implemented in a way thats readable ( /post/:siteId )
Page.js does this logic one-way. If we have a path in page, I have to do some regec magic to get path without site. I can do that, but it will be less consistent (I would like to keep consistency between analytics and error logging)